From 370f5d4d76e2922089c152b2ede97f922f8fdaa8 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 3 Mar 2016 19:11:08 +0200
Subject: [PATCH 0936/1240] tools: a3700: Add clocks setup to XDB DDR script

- Add clocks configuration to DDR init script for XDB
  This change is requred for running SPL from DDR and
  not from SRAM.
- The clocks are using 600-600 preset.

Change-Id: I326966983f210e304b939e38462c28842d75ce3e
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28036
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
---
 tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb | 74 +++++++++++++++++++++++-
 1 file changed, 73 insertions(+), 1 deletion(-)

diff --git a/tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb b/tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb
index 89abe78..f2749e0 100644
--- a/tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb
+++ b/tools/marvell/debugger_scripts/XDB/a3700_ddr.xdb
@@ -8,8 +8,80 @@
 ! UPDATE NOR TO KEEP CURRENT THE INFORMATION CONTAINED IN THIS DOCUMENT.
 ! -----------------------------------------------------------------------
 !
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+! Switch all clocks to REFCLOCK
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+SET VALUE /SIZE=LONG CORE:1(0xD0013010)=0x00000000
+SET VALUE /SIZE=LONG CORE:1(0xD0018010)=0x00000000
+
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+! TBG-A: SE vco_div)=0x0,
+! DIFF vco_div)=0x1, vco_range)=0xd
+! tbg_N)=0x48 KVCO = 2400 MHz
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+SET VALUE /SIZE=LONG CORE:1(0xD0013204)=0x00C00091
+SET VALUE /SIZE=LONG CORE:1(0xD0013204)=0x00C00121
+SET VALUE /SIZE=LONG CORE:1(0xD0013220)=0x08030803
+SET VALUE /SIZE=LONG CORE:1(0xD0013208)=0x94011401
+SET VALUE /SIZE=LONG CORE:1(0xD0013230)=0x00020002
+SET VALUE /SIZE=LONG CORE:1(0xD0013208)=0x94011401
+SET VALUE /SIZE=LONG CORE:1(0xD001320C)=0x53E556E6
+SET VALUE /SIZE=LONG CORE:1(0xD0013210)=0x014A014D
+SET VALUE /SIZE=LONG CORE:1(0xD001320C)=0x53E556E6
+SET VALUE /SIZE=LONG CORE:1(0xD0013204)=0x00C00120
+SET VALUE /SIZE=LONG CORE:1(0xD0013208)=0x94011401
+
+DEFINE SYMBOL /TYPE="unsigned int" /ADDRESS=0xD0013208 TestVal
+
+WHILE TestVal & 0x80008000 != 0x80008000 THEN
+    TCI DELAY 1
+END
+TCI DELAY 1
+
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+! TBG-B: SE vco_div)=0x1,
+! DIFF vco_div)=0x1, vco_range)=0xb
+! tbg_N)=0x3c KVCO = 2000 MHz
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+SET VALUE /SIZE=LONG CORE:1(0xD0013204)=0x00C10120
+SET VALUE /SIZE=LONG CORE:1(0xD0013204)=0x00F10120
+SET VALUE /SIZE=LONG CORE:1(0xD0013220)=0x08030803
+SET VALUE /SIZE=LONG CORE:1(0xD0013208)=0x14019401
+SET VALUE /SIZE=LONG CORE:1(0xD0013230)=0x00020002
+SET VALUE /SIZE=LONG CORE:1(0xD0013208)=0x14019401
+SET VALUE /SIZE=LONG CORE:1(0xD001320C)=0x56E556E6
+SET VALUE /SIZE=LONG CORE:1(0xD0013210)=0x014B014D
+SET VALUE /SIZE=LONG CORE:1(0xD001320C)=0x56E656E6
+SET VALUE /SIZE=LONG CORE:1(0xD0013204)=0x00F00120
+SET VALUE /SIZE=LONG CORE:1(0xD0013208)=0x14019401
+
+WHILE TestVal & 0x80008000 != 0x80008000 THEN
+    TCI DELAY 1
+END
+TCI DELAY 1
+
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+! Set clocks to 600/600 preset
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+SET VALUE /SIZE=LONG CORE:1(0xD0013014)=0x00000000
+SET VALUE /SIZE=LONG CORE:1(0xD0013004)=0x2326202A
+SET VALUE /SIZE=LONG CORE:1(0xD0013008)=0x1A09AAA9
+SET VALUE /SIZE=LONG CORE:1(0xD001300C)=0x208B3482
+SET VALUE /SIZE=LONG CORE:1(0xD0013000)=0x0333C0FE
+SET VALUE /SIZE=LONG CORE:1(0xD0018014)=0x00180000
+SET VALUE /SIZE=LONG CORE:1(0xD0018004)=0x053154C8
+SET VALUE /SIZE=LONG CORE:1(0xD0018008)=0x00307880
+SET VALUE /SIZE=LONG CORE:1(0xD001800C)=0x00000940
+SET VALUE /SIZE=LONG CORE:1(0xD0018000)=0x003F8F40
+SET VALUE /SIZE=LONG CORE:1(0xD0013210)=0x0014B014D
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+! Switch all clocks to back dividers
+!+++++++++++++++++++++++++++++++++++++++++++++++++
+SET VALUE /SIZE=LONG CORE:1(0xD0013010)=0x00009FFF
+SET VALUE /SIZE=LONG CORE:1(0xD0018010)=0x000007AA
+
+!+++++++++++++++++++++++++++++++++++++++++++++++++
 ! Running DDR initialization in 600MHz mode
-! All system clocks are set to default by the HW
 !+++++++++++++++++++++++++++++++++++++++++++++++++
 SET VALUE /SIZE=LONG CORE:1(0xD0014008)=0x00404500
 SET VALUE /SIZE=LONG CORE:1(0xD0002000)=0x00010000
-- 
1.9.1

